tests/test_client_credential.py

```diff
@@ -55,35 +57,28 @@ def tearDown(self):
 class TestClientCredential(BaseTest):
     def test_client_credential_access_allowed(self):
         """
-        Request an access token using Client Credential Flow
+        Request an access token using Client Credential Flow with hashed secrets
         """
+        self.assertNotEqual(self.application.client_secret, CLEARTEXT_SECRET)
+
         token_request_data = {
             "grant_type": "client_credentials",
         }
-        auth_headers = get_basic_auth_header(self.application.client_id, self.application.client_secret)
+        auth_headers = get_basic_auth_header(self.application.client_id, CLEARTEXT_SECRET)
 
         response = self.client.post(reverse("oauth2_provider:token"), data=token_request_data, **auth_headers)
         self.assertEqual(response.status_code, 200)
 
-        content = json.loads(response.content.decode("utf-8"))
-        access_token = content["access_token"]
-
-        # use token to access the resource
-        auth_headers = {
-            "HTTP_AUTHORIZATION": "Bearer " + access_token,
-        }
-        request = self.factory.get("/fake-resource", **auth_headers)
-        request.user = self.test_user
-
-        view = ResourceView.as_view()
-        response = view(request)
-        self.assertEqual(response, "This is a protected resource")
+        # secret mismatch should return a 401
+        auth_headers = get_basic_auth_header(self.application.client_id, "not-the-secret")
+        response = self.client.post(reverse("oauth2_provider:token"), data=token_request_data, **auth_headers)
```

Comment:
I'd still prefer using RequestFactory for speed of testing and more manageable configuration. Only reason I'd use client is for middleware; is there middleware enabled here?

If we're testing for a 401 through an exception, just use `pytest.raises`?

Reply:
My goal was to do the minimum needed to get what was started and reverted in #1020 simplified and merged. Cleaning up these tests and making them consistent is a much larger issue that I'm not comfortable with tackling; there are something like 3 different styles used.

A (series of) PR(s) addressing refactoring the test suite would be appreciated!